libraries/Microsoft.Bot.Builder/Inspection/InspectionSessionsByStatus.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    public class InspectionSessionsByStatus
+    {
+        public IDictionary<string, ConversationReference> OpenedSessions { get; } = new Dictionary<string, ConversationReference>();
+
```

Comment:
BUG: These need to be concurrent dictionaries. You add to them, and query them, from multiple threads.

Reply:
These are only queried from a single thread. The external storage is the way things are shared.

However, the storage is not locked and so technically one can stamp on another; this uses the current model so inherits the limitations.

But that doesn't change this.